hepdata/modules/records/views.py

```diff
@@ -699,7 +713,13 @@ def get_resource(resource_id):
                 print("Resource is at: " + resource_obj.file_location)
                 try:
                     with open(resource_obj.file_location, 'r', encoding='utf-8') as resource_file:
-                        contents = resource_file.read() if mimetypes.guess_type(resource_obj.file_location)[0] != 'application/x-tar' else 'Binary'
+                        if mimetypes.guess_type(resource_obj.file_location)[0] != 'application/x-tar':
+                            # Check against the filesize threshold. Do not set contents if it fails.
+                            filesize = os.path.getsize(resource_obj.file_location)
+                            if filesize < ADDITIONAL_SIZE_LOAD_CHECK_THRESHOLD:
+                                contents = resource_file.read()
```

Comment:
For a large `filesize`, the `contents = ''` (an empty string).  Maybe change to setting `contents = 'Large text file'`, which would be more informative if viewing the JSON format of the landing page.  The code in `resource_details.html` would then need to be modified.

Note that the [comment](https://github.com/hepdata/hepdata/issues/136#issuecomment-1611505709) in the original issue mentioned modifications to `hepdata_code_view.js` as well as `resource_details.html`.  The former is used for displaying resource files attached to individual tables rather than the whole submission.